fix: properly decoded non-indexed reference types

ju61n opened the pull request:
Non-indexed references in events where improperly being handled. Only indexed reference types are not actually stored in event data and 32 byte hash is stored instead.

ju61n committed:
fix: properly decoded non-indexed reference types

Non-indexed references in events where improperly being handled. Only
indexed reference types are not actually stored in event data and 32
byte hash is stored instead.

trv-cs commented:
Could you please add some tests to check these cases, ideally we need tests for:

- [x] No reference types
- [x] Non-Indexed reference types
- [x] Indexed reference types
- [x] Mixed Indexed and Non-Indexed reference types

ju61n committed:
test: add tests to cover event decoding changes

This commits adds four test to cater for the different possible
configurations of event data mention by Tony in the PR.

Events can have indexed and non-indexed data, and reference and
non-reference data. Mixing the two raises some possibly unexpected
behaviour. Specifically, the indexed reference types are lossy that is
they lose their original data in favour of hash.

trv-cs commented on tests/test_decode.py:
```diff
@@ -278,6 +277,91 @@ def test_signature(
+                    "inputs": [
+                        {"indexed": False, "name": "test", "type": "string"},
+                    ],
+                    "name": "NonIndexedRefernce",
```
Typo `NonIndexedRefernce`

ju61n committed:
fix: NonIndexedRefernce -> NonIndexedReference
